### PR TITLE
MEN-5459 Install go based on the source repository preferences.

### PR DESCRIPTION
**Pull request by merlin-northern**
ChangeLog:title
Signed-off-by: Peter Grzybowski <peter@northern.tech>

### CODE REVIEW

**Comment by merlin-northern** [comment]
@lluiscampos requesting early review, does it make any sense?

**Comment by merlin-northern** [comment]
@lluiscampos goes with:
* https://github.com/mendersoftware/mender-gateway/pull/22
* https://github.com/mendersoftware/mender-connect/pull/62
* https://github.com/mendersoftware/mender/pull/958

File: mender-deb-package [hunk 1]
```diff
@@ -50,6 +50,24 @@ checkout_repo() {
...
+ golang_version_set=${golang_version_set//\"/}
+ GOLANG_VERSION=${golang_version_set:-${GOLANG_VERSION}}
+
+ wget -q https://dl.google.com/go/go$GOLANG_VERSION.linux-amd64.tar.gz \
+    && tar -C /usr/local -xzf go$GOLANG_VERSION.linux-amd64.tar.gz
+ export GOPATH="/root/go"
+ export PATH="$PATH:/usr/local/go/bin"
+ # Support building mender-client 2.3.x, since it does not have go modules support
+ # For newer clients with a go.mod file, this is a no-op however.
+ export GO111MODULE=auto
```
Comment (lluiscampos) [inline mender-deb-package:67]: This we can remove, really. 2.3.x has been EOL for a long while now

File: mender-deb-package [hunk 2]
```diff
@@ -50,6 +50,24 @@ checkout_repo() {
   git checkout $VERSION
 }
 
+install_go() {
+ local GOLANG_VERSION=1.14.7
```
Comment (lluiscampos) [inline mender-deb-package:54]: I think default should be `1.17.6`, which is what we currently have for all.

File: mender-deb-package [hunk 3]
```diff
@@ -50,6 +50,24 @@ checkout_repo() {
...
+ local golang_version_set
+
+ golang_version_set=$(sed -n '/[^ ]*GOLANG_VERSION:[ "0-9]/{s/.*GOLANG_VERSION:[ ]*\(["0-9\.]*\).*/\1/p;q}' .gitlab-ci.yml)
+ golang_version_set=${golang_version_set//\"/}
+ GOLANG_VERSION=${golang_version_set:-${GOLANG_VERSION}}
+
+ wget -q https://dl.google.com/go/go$GOLANG_VERSION.linux-amd64.tar.gz \
```
Comment (lluiscampos) [inline mender-deb-package:61]: Will this work? In the other PRs we are setting `GOLANG_VERSION` to `1.17` (missing bugfix number).

Comment (lluiscampos) [reply mender-deb-package:61]: It works :+1:

**Review by lluiscampos — APPROVED** [review]

File: mender-deb-package (hunk 2 above)
Comment (merlin-northern) [reply mender-deb-package]: changed.

File: mender-deb-package (hunk 1 above)
Comment (merlin-northern) [reply mender-deb-package]: removed.

File: mender-deb-package [hunk 4]
```diff
@@ -50,6 +50,23 @@ checkout_repo() {
...
+    && tar -C /usr/local -xzf go$GOLANG_VERSION.linux-amd64.tar.gz
+ export GOPATH="/root/go"
+ export PATH="$PATH:/usr/local/go/bin"
+ # Support building mender-client 2.3.x, since it does not have go modules support
+ # For newer clients with a go.mod file, this is a no-op however.
```
Comment (lluiscampos) [inline mender-deb-package:66]: Please remove the comment also

Comment (merlin-northern) [reply mender-deb-package]: removed.

**Pull request merged**